cmd/podman/system/prune.go

```diff
@@ -73,32 +73,55 @@ Are you sure you want to continue? [y/N] `, volumeString)
 			return nil
 		}
 	}
-	// TODO: support for filters in system prune
-	response, err := registry.ContainerEngine().SystemPrune(context.Background(), pruneOptions)
-	if err != nil {
-		return err
-	}
-	// Print pod prune results
-	fmt.Println("Deleted Pods")
-	err = utils.PrintPodPruneResults(response.PodPruneReport)
-	if err != nil {
-		return err
-	}
-	// Print container prune results
-	fmt.Println("Deleted Containers")
-	err = utils.PrintContainerPruneResults(response.ContainerPruneReport)
-	if err != nil {
-		return err
-	}
-	// Print Volume prune results
-	if pruneOptions.Volume {
-		fmt.Println("Deleted Volumes")
-		err = utils.PrintVolumePruneResults(response.VolumePruneReport)
+	for {
```

Comment:
Recommendation: We should put a max iteration count on this - if we haven't removed everything in, say, 25 loops, we probably have a problem.

Reply:
Just a quick note to agree, but suggest a larger number: I've seen 40-odd manual iterations before… so set the bar at 50?

Reply:
Yeah, I'm not at all a fan of a naked for loop.

Reply:
If each iteration takes a beat, we might want to add a message like "Working" that adds a dot after each iteration or after some number of iterations.

Reply:
Each successful interaction will print purged lines, So no need for heartbeat.